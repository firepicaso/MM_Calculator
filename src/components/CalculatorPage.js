import Calculator from "./Calculator";

export default function CalculatorPage() {
    return (
        <div>
            <calculator />
        </div>
    )
};