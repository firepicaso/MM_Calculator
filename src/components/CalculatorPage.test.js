import CalculatorPage from "./CalculatorPage";
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

test('testing CalculatorPage component', () => {
    const { container } = render(<CalculatorPage />);
    expect(container.firstChild).toMatchSnapshot();
  });
  
  test('simulating user interaction', () => {
    render(<CalculatorPage />);
  
    const buttonElement = screen.getByText('2');
    fireEvent.click(buttonElement);
  
    const resultElements = screen.queryAllByText('2');
    const resultElement = resultElements.find((element) => element.tagName === 'DIV');
    expect(resultElement).toBeInTheDocument();
});