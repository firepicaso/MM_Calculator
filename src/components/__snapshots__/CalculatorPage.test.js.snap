// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CalculatorPage component 1`] = `
<div>
  <div
    class="app-container"
    data-testid="calculator-component"
  >
    <h2>
      Let's do some math!
    </h2>
    <div
      class="calculator"
    >
      <div
        class="display"
        data-testid="display"
      >
        0
      </div>
      <div
        class="button-row"
      >
        <button
          class=""
          type="button"
        >
          AC
        </button>
        <button
          class=""
          type="button"
        >
          +/-
        </button>
        <button
          class=""
          type="button"
        >
          %
        </button>
        <button
          class=""
          type="button"
        >
          รท
        </button>
      </div>
      <div
        class="button-row"
      >
        <button
          class=""
          type="button"
        >
          7
        </button>
        <button
          class=""
          type="button"
        >
          8
        </button>
        <button
          class=""
          type="button"
        >
          9
        </button>
        <button
          class=""
          type="button"
        >
          x
        </button>
      </div>
      <div
        class="button-row"
      >
        <button
          class=""
          type="button"
        >
          4
        </button>
        <button
          class=""
          type="button"
        >
          5
        </button>
        <button
          class=""
          type="button"
        >
          6
        </button>
        <button
          class=""
          type="button"
        >
          -
        </button>
      </div>
      <div
        class="button-row"
      >
        <button
          class=""
          type="button"
        >
          1
        </button>
        <button
          class=""
          type="button"
        >
          2
        </button>
        <button
          class=""
          type="button"
        >
          3
        </button>
        <button
          class=""
          type="button"
        >
          +
        </button>
      </div>
      <div
        class="button-row bottom-row"
      >
        <button
          class=""
          type="button"
        >
          0
        </button>
        <button
          class="narrow"
          type="button"
        >
          .
        </button>
        <button
          class="narrow"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
